_posts/2017-07-06-hitchhikers-guide-to-cp.md

```diff
@@ -0,0 +1,308 @@
+
+# Common Mistakes and Errors
+
+"Anyone who has never made a mistake has never tried anything new."      
+
+-- Albert Einstein
+
+Following are few frequently encountered mistakes and errors while writing codes--
+
++   **Not including header file for a function or data structure.** Results in compilation error.
+
++   **Not initializing variables.** While calculating values like sum or factorial, using uninitialized variables can result in wrong answer.
+
++   **Declaring arrays of size smaller than required.** This results in Runtime Error. Check if array size is enough for all possible test cases and increase it, if required.
+
++   **Using int in place of long long and float in place of double.** Using int when an integer having value greater than $$2 * 10^9$$ is to be stored will cause an integer overflow and program will give Wrong answer. Use long long when integer is upto $$10^{18}$$. Double is having more precision as compared to float and hence, must be prefered over float.
+
++   **Comparing floats or doubles.** Float and double do not have infinite precision. One can't just directly check for two floating point values for equality.
+
+Instead of
+```
+double a, b;
+scanf("%lf%lf", &a, &b);
+if(a == b) {
+    //statements
+}
+```
+
+Use
+```
+double a, b;
+scanf("%lf%lf", &a, &b);
+if(fabs(a - b) < 0.000001) {
+    //statements
+}
+```
+
+**fabs()** calculates the absolute diff in a and b. If values are equal, diff will be negligible and above condition will be true.
+
+---
+
+# Some Pro Tips 
+
+Following are few points that one can make use of in order to get better at Competitive programming--
+
++   Get yourself familiar with well-known algorithms and data structures. Read tutorials, watch video lectures and try to implement them in your own way.
+
++   PRACTICE is the mantra to being a better programmer. Practice problems on SPOJ, Codeforces, Codechef or any other judge whenever you find time. Ignore doing problems that are easy for you, instead do those problems which are challenging for you.
+
++   Don't just do problems using algorithm but instead understand how the given algorithm works and you'll be able to modify it according to the question's requirements.
+
++   Do read the editorials after the contest is over. Try to understand the approach of those problems that you were unable to do during the contest. After that, try solving them. It'll help you understand new concepts and their implementation.
+
++   There's no harm in seeing other's code after the contest as long as you are not doing *copy-paste*. You'll learn how to implement codes in simple and elegant way.
+
++   It's ok to spend hours on coming up with an approach to a problem. But, if you are stuck for too long, discuss the problem with friends and within your groups. Someone might have already done the problem or might help you in coming up with an approach.
+
++   Most importantly, you need to be persisitive and keep on going and you'll soon be a pro :). 
```

Comment:
`persisitive` -> `persistent` or `persistive`. I personally prefer `persistent` because of the opportunity for puns there ;)